js/foam/u2/search/FilterController.js

```diff
@@ -0,0 +1,451 @@
+  traits: [
+    'foam.u2.md.MDViewOverridesTrait',
+  ],
+
+  requires: [
+    'foam.u2.TextField',
+    'foam.u2.md.Card',
+    'foam.u2.md.Select',
+    'foam.u2.md.TableView',
+    'foam.u2.search.GroupBySearchView',
+    'foam.u2.search.SearchMgr',
+    'foam.u2.search.TextSearchView'
+  ],
+
+  imports: [
+    'title$',
+    'window'
+  ],
+
+  exports: [
+    'as filterController',
+  ],
+
+  properties: [
+    {
+      name: 'count',
+    },
+    {
+      name: 'totalCount',
+    },
+    {
+      name: 'model',
+      defaultValueFn: function() { return this.data.model; }
+    },
+    {
+      name: 'data',
+      required: true,
+      postSet: function(old, nu) {
+        if (nu) {
+          nu.select(COUNT())(function(c) {
+            this.totalCount = c.count;
+          }.bind(this));
+        }
+      }
+    },
+    {
+      model_: 'foam.core.types.DAOProperty',
+      name: 'filteredDAO',
+      required: true,
+      toPropertyE: function(X) {
+        return X.lookup('foam.u2.md.TableView').create({
+          editColumnsEnabled: true,
+        }, X);
+      }
+      //REMOVE
+      //view: {
+      //  factory_: 'foam.ui.md.FlexTableView',
+      //  editColumnsEnabled: true,
+      //  rowHeight: 70,
+      //  scrollEnabled: true
+      //}
+    },
+    {
+      name: 'filterChoice',
+      label: 'New Filter',
+    },
+    {
+      name: 'title',
+    },
+    {
+      name: 'defaultFilter',
+      documentation: 'Overridden by the specific subtypes.',
+    },
+    {
+      name: 'searchMgr',
+      lazyFactory: function() {
+        return this.SearchMgr.create({
+          dao$: this.data$,
+          filteredDAO$: this.filteredDAO$,
+        });
+      }
+    },
+    {
+      model_: 'StringArrayProperty',
+      name: 'searchFields',
+      documentation: 'Property names that are currently selected as filters.',
+      postSet: function(old, nu) {
+        // Build up objects of the keys demanded by both sides, then add and
+        // remove as needed.
+        if (old) {
+          for (var i = 0; i < old.length; i++) {
+            if (!nu || nu.indexOf(old[i]) < 0) {
+              this.searchMgr.remove(old[i]);
+              this.searchViews[old[i]].remove();
+              delete this.searchViews[old[i]];
+            }
+          }
+        }
+
+        if (nu) {
+          for (var i = 0; i < nu.length; i++) {
+            if (!old || old.indexOf(nu[i]) < 0) {
+              var split = this.splitName(nu[i]);
+              var prop = this.model.getFeature(split.name);
+              var model = BooleanProperty.isInstance(prop) ?
+                  this.GroupBySearchView : this.TextSearchView;
+              var options = {
+                floatingLabel: false,
+                name: nu[i]
+              };
+              if (prop.tableSeparator) {
+                options.split = prop.tableSeparator;
+              }
+              this.addGroup(model, prop, options);
+              this.renderFilter(nu[i]);
```

Comment:
Seems like a race condition.

We should guard this on this.filtersE_ being set, or put a factory on this.filtersE_ so that we don't die if this postSet happens before initE.